File: seungtae/23_09/programmers/17686.java
import java.util.*;

public class Solution {
	public String[] solution(String[] files) {
		String[] answer = {};

		Arrays.sort(files, new Comparator<String>() {
			@Override
			public int compare(String o1, String o2) {
				String[] file1 = split(o1);
				String[] file2 = split(o2);

				int isSameHead = file1[0].compareTo(file2[0]);

				if(isSameHead == 0) {
					int num1 = Integer.parseInt(file1[1]);
					int num2 = Integer.parseInt(file2[1]);

					return num1 - num2;
				} else {
					return isSameHead;
				}
			}

			private String[] split(String str){
				String header = "";
				String number = "";
				String footer = "";

				int i = 0;
				while (i < str.length() && !Character.isDigit(str.charAt(i))) {
					header += str.charAt(i);
					i++;
				}

				while (i < str.length() && Character.isDigit(str.charAt(i))) {
					number += str.charAt(i);
					i++;
				}

				footer = str.substring(i);

				String[] file = { header.toLowerCase(), number, footer };

				return file;
			}
		});

		for (String file: files) {
			System.out.println("file = " + file);
		}

		return files;
	}


	public static void main(String[] args) {
		Solution s = new Solution();
		System.out.println(s.solution(new String[] { "img12.png", "img10.png", "img02.png", "img1.png", "IMG01.GIF", "img2.JPG" }));
//		s.solution(new String[] { "img12.png", "img10.png", "img02.png", "img1.png", "IMG01.GIF", "img2.JPG" });
//		s.solution(new String[] { "F-5 Freedom Fighter", "B-50 Superfortress", "A-10 Thunderbolt II", "F-14 Tomcat" });

	}
}

File: kwonmin/23_09/programmers/17686.py
# 좀 귀찮은 문제가 있네
# 원본 그대로 출력해야 한다는 문제...
# 아스키로 하면 어떨까? 라는 생각이 들었지만
# 10만 범위라서 곤란할듯?
# 그럼 숫자만 비교를 해야 하는데


def divide(x):
    head = ''
    number = ''
    tail = ''

    start = 0
    end = len(x)
    flag = True
    for i in range(len(x)):
        if x[i].isnumeric():
            if not start:
                flag = False
                start = i
        
        elif not flag:
            end = i
            break
    
    head = x[:start]
    number = x[start:end]
    tail = x[end:]
    return (head, number, tail)

def solution(files):
    answer = []
    temp = []
    for file in files:
        temp.append(divide(file))

    temp.sort(key=lambda x: (x[0].lower(), int(x[1])))
    for a,b,c in temp:
        answer.append(a+b+c)
    return answer

print(solution(["img12", "img10", "img02", "img1", "IMG01.GIF", "img2.JPG"]))
